draft-ietf-rtcweb-security-arch.xml

```diff
@@ -556,6 +556,163 @@
+           verification fails, it MUST reject the offer.
+         </t>
+         </section>
+         <section title="Generating of SDP Answer" anchor="sec.sdp-id-attr-oa-ansa">
+         <t>
+           When an answerer sends an answer, in order to provide its identity
+           assertion to the peer, it includes an 'identity' attribute in the
+           answer. In addition, it MUST include 'fingerprint' attributes for
+           each fingerprint that have been used to assert the identity.
+         </t>
+         <t>
+           NOTE: The answerer can include an 'identity' attribute in the answer
+           independently of whether the corresponding offer contained an
+           identity attribute or not.
+         </t>
+         </section>
+         <section title="Offerer Processing of SDP Answer" anchor="sec.sdp-id-attr-oa-offa">
+         <t>           
+           When the offerer receives an answer that contains an 'identity'
+           attribute, the offerer can use the attribute information to contact
+           the IdP, and verify the identity of the peer. If the offerer does
+           not accept the attribute (or the attribute value), or if the identity
+           verification fails, it MUST take proper actions to reject the answer.
+         </t>
+         </section>
+         <section title="Modifying Session" anchor="sec.sdp-id-attr-oa-modi">
+         <t>
+           When an endpoint has provided its identity assertion in a previous
+           successful offer/answer exchange, and when the endpoint later in the 
+           session sends an offer or answer, the endpoint can either keep the 
+           identity assertion value unchanged, modify the value, or revoke the 
+           identity assertion. This section describes the procedures for doing so.
+         </t>
+         <t>
+           If the endpoint wants to maintain the previously provided identity assertion, 
+           it MUST include an identity attribute with the previously provided attribute value
+           in the offer or answer. In addition, it MUST include 'fingerprint' attributes for
+           each fingerprint that have been used to assert the identity.
+         </t>
+         <t>     
+           If the endpoint wants to modify its identity assertion within the
+           session, it MUST include an 'identity' attribute value with the new
+           identity assertion in the offer or answer. In addition, it MUST include
+           'fingerprint' attributes for each fingerprint that have been used to
+           assert the identity.
+         </t>
+         <t>
+           If the endpoint wants to revoke its identity assertion from the
+           session, it MUST NOT include an identity attribute in the offer or answer.
+         </t>
+         <t>
+           NOTE: Specific usages might restrict the modification and/or revoking
+           of the identity assertion within a session.
+         </t>
+         </section>         
```

Comment:
Add:
Endpoints can add, change, or remove 'fingerprint' attributes when modifying a session.  If a modified session retains the 'identity' attribute, there MUST be at least one 'fingerprint' attribute that applies to each media section and any 'fingerprint' attributes that are included MUST be covered by the 'identity' attribute.

Wording negotiable.

Reply:
Will add.

Reply:
I mean sure it's covered, but I do not think that it will hurt to include this text to make it crystal clear that if you're modifying a session you need to do what it says.

Reply:
It would be repeating the same thing, using different terminology. I don't think that clarifies anything :)

But, if there is something that is not covered in the existing text I am happy to add that. I just can't find what it is.

Reply:
There are four things being covered in this section:
1. What to do when not changing the identity attribute.
2. What to do when changing the identity attribute.
3. What to do when changing the fingerprint attribute.
4. How to revoke the identity.

The add Martin is asking for addresses #3.  And, not the sentence you quotes is also in the paragraph addressing #1.

Reply:
The text says that there must be a fingerprint attribute for each fingerprint that has been used to assert the identity. That is all that matters as far as this draft is concerned. 

Modifying one or more fingerprint attributes so that rule no longer apply is a protocol error.

Reply:
Happy to say instead that a=fingerprint can be added, changed, or removed as long as the session description as a whole remains consistent with these rules.

Reply:
One question: is it allowed to include fingerprints that have NOT been used to assert the identity? If so, then it is possible to modify attributes without impacting the Identity attribute.

But, if one or more of the fingerprints that are used to assert the identity change, I assume the Identity attribute will also change as a result?